crates/exec-wasmtime/src/runtime/net/tls.rs

```diff
@@ -130,6 +131,7 @@ impl Stream {
         let mut stream = Self {
             tcp,
             tls,
+            nonblocking: false, // this is only valid under assumption that this executable has opened the socket
```

Comment:
you can query the current state of blocking with libc::fcntl(fd, libc::F_GETFL...) and check for O_NONBLOCK.

Reply:
Right, but we're opening the socket ourselves, why do a syscall?

Reply:
ok